proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+...
+
+Project 'MyProjectB' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+   > Microsoft.Win32.Primitives                                                         4.3.0
+...
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+```
+
+### dotnet list package --include-transitive --json --flatten
+
+```json
+{
+  "version": 1,
+  "topLevelPackages": [
+       {
+            "id": "Microsoft.Extensions.Primitives",
+            "requestedVersion": "[1.0.0, 5.0.0]",
+            "resolvedVersion": "1.0.0"
+        },
+        {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "[1.1.2, 2.0.0)",
+            "resolvedVersion": "1.1.2"
+        }
+    ],
+    "transitivePackages": [
+        {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+        },
+        {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+        },
+        {
+            "id": "Microsoft.NETCore.Targets",
+            "resolvedVersion": "1.1.0"
+        }
+...
+    ]
+}
+```
+
+### Compatibility
+
+ We start with `version 1`, as long as we don't remove or rename then it'll be backward compatible. In case [we change version](https://stackoverflow.com/a/13945074) just add new properties, keep old ones even it's not used.
+
+### Out-of-scope
+
+* To avoid disk I/O, we won't support saving the machine-readable output to disk as part of this spec. The work-around is for the consumer to read from the console's stdout stream.
+* At this point, no other CLI commands (e.g. dotnet list reference) will be within scope for this feature.
+"--parsable" option needs separate spec.
```

Comment:
It looks it's more it's concise, [cli manipulation friendly version](https://stackoverflow.com/a/31586321). 
```
npm ls --parseable
C:\NuGetProj\IssueRepro\JsonOutput
C:\NuGetProj\IssueRepro\JsonOutput\node_modules\http-proxy
C:\NuGetProj\IssueRepro\JsonOutput\node_modules\eventemitter3
C:\NuGetProj\IssueRepro\JsonOutput\node_modules\requires-port
C:\NuGetProj\IssueRepro\JsonOutput\node_modules\lodash
C:\NuGetProj\IssueRepro\JsonOutput\node_modules\websocket-extensions

```
 ```
npm ls --json
{
  "dependencies": {
    "http-proxy": {
      "version": "1.16.2",
      "from": "http-proxy@1.16.2",
      "resolved": "https://registry.npmjs.org/http-proxy/-/http-proxy-1.16.2.tgz",
      "dependencies": {
        "eventemitter3": {
          "version": "1.2.0",
          "from": "eventemitter3@1.x.x",
          "resolved": "https://registry.npmjs.org/eventemitter3/-/eventemitter3-1.2.0.tgz"
        },
        "requires-port": {
          "version": "1.0.0",
          "from": "requires-port@1.x.x",
          "resolved": "https://registry.npmjs.org/requires-port/-/requires-port-1.0.0.tgz"
        }
      }
    },
    "lodash": {
      "version": "4.2.0",
      "from": "lodash@4.2.0",
      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.2.0.tgz"
    },
    "websocket-extensions": {
      "version": "0.1.1",
      "from": "websocket-extensions@0.1.1",
      "resolved": "https://registry.npmjs.org/websocket-extensions/-/websocket-extensions-0.1.1.tgz"
    }
  }
}
```

Reply:
I selected `json` implementation over `parseable` due to many favorable comments (https://github.com/nuget/home/issues/7752#issuecomment-459836491,
https://github.com/nuget/home/issues/7752#issuecomment-459907122,
https://github.com/nuget/home/issues/7752#issuecomment-474191608,
https://github.com/nuget/home/issues/7752#issuecomment-5129720081). 
Of course `parseable` is still option next time, but I felt it's too much for this spec.

Reply:
By the way another candidate in the future is `yaml`

Reply:
YAML is OK for files edited by people, but I wouldn't want it for data that is generated by one program and intended to be parsed by another, because it needs a more complex parser than JSON.